yt/frontends/gizmo/data_structures.py

```diff
@@ -38,9 +42,16 @@ def _is_valid(cls, filename, *args, **kwargs):
             valid = all(ng in fh["/"] for ng in need_groups) and not any(
                 vg in fh["/"] for vg in veto_groups
             )
-            dmetal = "/PartType0/Metallicity"
-            if dmetal not in fh or fh[dmetal].shape[1] < 11:
-                valid = False
+            # From Apr 2021, 7f1f06f, public gizmo includes a header variable
+            # GIZMO_version, which is set to the year of the most recent commit
+            # We should prefer this to checking the metallicity, which might
+            # not exist
+            if "GIZMO_version" not in fh["/Header"].attrs:
+                dmetal = "/PartType0/Metallicity"
+                if dmetal not in fh or (
+                    fh[dmetal].ndim > 1 and fh[dmetal].shape[1] < 11
+                ):
+                    valid = False
```

Comment:
Can I get a second opinion on this @neutrinoceros @jzuhone ? I ran into this while generating a very simple sample dataset which doesn't for example have cooling/metallicity turned on (the basic zeldovich pancake test) but wasn't sure this is actually something to worry about.

Reply:
I'm not sure I understand the question here. Are you worried you might accidentally invalidate gizmo files ?

Reply:
I suppose, yes, but more that I wasn't sure that checking for GIZMO_version was sufficient. Basically, if we have GIZMO_version, is that all we need to say this is a valid gizmo file? Or should it check for something else (like dmetal) as well?

Reply:
> Basically, if we have GIZMO_version, is that all we need to say this is a valid gizmo file?

I suppose it should be a sufficient (but not necessary) validator, yes.
What's important here is that this function returns `True` for any known gizmo dataset (which if I'm reading your first comment correctly isn't the case at the moment ?) and `False` for every known non-Gizmo dataset, where by "known" I mean "datasets that are used in CI + any additional dataset you can get your hands on". Basically if this ever returns `False` for a datafile you know is from GIZMO, it's a bug :)